File: 2571-FindThePivotInteger/2571-FindThePivotInteger.py
# Last updated: 5/29/2026, 11:52:01 AM
class Solution:
    def pivotInteger(self, n: int) -> int:
        t=n*(n+1)//2
        l=0
        for i in range(1,n+1):
            t-=i
            if l==t:
                return i
            l+=i
        return -1

File: 2571-FindThePivotInteger/2571-FindThePivotInteger.java
// Last updated: 5/29/2026, 11:52:07 AM
class Solution {
    public int pivotInteger(int n) {
        int sum=0;
        for(int i=1;i<=n;i++){
            sum=sum+i;
        }
        int su=0;
        for(int i=n;i>0;i--){
            su=su+i;
            if(sum==su){
                return i;
            }
            sum=sum-i;

        }
        return -1;
    }
}